### PR TITLE
Edit usage documentation

### PR DESCRIPTION
**Pull request by willingc**
- Make the doc more user focused
- Add a bit more text for understanding
- Image can be replaced at a later date.

### CODE REVIEW

File: docs/source/usage.rst [hunk 1]
```diff
@@ -1,47 +1,75 @@
...
 
-Every *save* of a notebook creates an *immutable copy* of the notebook on object storage.
+                                             *store*
 
-To simplify implementation, we currently rely on S3 as the object store, using
+Data scientist saves to Local System ------------------> Remote Data Store (i.e. S3) Workspace
+
+                                             *clone*
+
+User requests a notebook to use locally <-------------- Remote Data Store (i.e. S3) Workspace
```
Comment (mpacer) [inline docs/source/usage.rst]: Tiny thing, just to be consistent, we should probably use Workspace to mean the semantics associated with workspace. Cloning can be done from any s3 path.

```suggestion
User requests a notebook to use locally <-------------- Remote Data Store (i.e. S3)
```

File: docs/source/usage.rst [hunk 2]
```diff
@@ -1,47 +1,75 @@
...
+
+How to store and clone versions
+-------------------------------
+
+Bookstore uses automatic notebook version management and specific storage paths when storing a notebook.
+
+Automatic Notebook Version Management
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Every *save* of a notebook creates an *immutable copy* of the notebook on object storage. Initially, Bookstore
```
Comment (mpacer) [inline docs/source/usage.rst]: While it is an immutable copy, I feel like we might want to be clearer that it is overwriting the file in place. It relies on versioned s3 buckets to handle the versioning.

File: docs/source/usage.rst [hunk 3]
```diff
@@ -1,47 +1,75 @@
...
-storage types:
+Bookstore archives notebook versions by keeping the path intact (until a user changes them). For example, the prefixes
+that could be associated with storage types:
 
 - Notebook in "draft" form: ``/workspace/kylek/notebooks/mine.ipynb``
 - Most recent published copy of a notebook: ``/published/kylek/notebooks/mine.ipynb``
 
-Scheduled notebooks will also be referred to by the notebook ``key``. In addition, we'll need to be able to surface
-version IDs as well.
+When scheduling execution of notebooks, each notebook path is a namespace that an external service can access. This
```
Comment (mpacer) [inline docs/source/usage.rst]: I actually wonder if we should remove this until we support some kind of hook into a scheduling system.

File: docs/source/usage.rst (hunk 1 above)
Comment (willingc) [reply docs/source/usage.rst]: No problem making the change. I want to make sure that I understand correctly how we want to use the word workspace.

My understanding: workspace is a path location to an S3 store where a user stores their notebooks.

File: docs/source/usage.rst (hunk 2 above)
Comment (willingc) [reply docs/source/usage.rst]: Good point on the overwrite. Will add.

File: docs/source/usage.rst (hunk 3 above)
Comment (willingc) [reply docs/source/usage.rst]: We can either 1) remove or 2) place in an information box that says that the current release does not support but we expect to do so in a later release. Let me know which you prefer. I don't have a strong feeling either way.

File: docs/source/usage.rst (hunk 2 above)
Comment (willingc) [reply docs/source/usage.rst]: Added to the following paragraph.

File: docs/source/usage.rst (hunk 3 above)
Comment (willingc) [reply docs/source/usage.rst]: Added as a note.

**Comment by rgbkrk** [comment]
Looks like review feedback is addressed. This is some additive improvement to the documentation; thank you Carol.

**Pull request merged**